.tekton/odh-trustyai-service-v2-16-pull-request.yaml

```
@@ -187,18 +223,131 @@ spec:
+      workspaces:
+      - name: output
+        workspace: workspace
+      - name: basic-auth
+        workspace: git-auth
+
+    - name: pnc-cli-build
+      runAfter:
+        - clone-pnc-cli-config-repository
+      taskSpec:
+        results:
+          - description: The file containing the list of pnc built artifacts to be later installed inside the container build
+            name: pnc-file-list
+        steps:
+          - name: run-pnc-build
+            image: quay.io/redhat-user-workloads/konflux-jbs-pnc-tenant/pnc/pnc-cli@sha256:00f29bb186e3afedbffeb57942038f7d9ad747640e0a4ae2c53c573a8cdece74
+            env:
+              - name: SSO_SERVICE_ACCOUNT_NAME
+                valueFrom:
+                  secretKeyRef:
+                    name: pnc-secret
+                    key: svc.account.username
+              - name: SSO_SERVICE_ACCOUNT_CLIENT_SECRET
+                valueFrom:
+                  secretKeyRef:
+                    name: pnc-secret
+                    key: svc.account.password
+              - name: PNC_PROFILE
+                valueFrom:
+                  secretKeyRef:
+                    name: pnc-secret
+                    key: pnc.profile
+              
+            script: |
+              #!/bin/bash
+              set -e
+
+              echo -e "\n=== Build config ==="
+              cat /workspace/output/build-config/$(params.build-config-path)
+              cp /workspace/output/build-config/$(params.build-config-path) /workspace/output
+              echo -e "===================="
+
+              echo -e "\nReplacing credentials in the CLI config ..."
+              envsubst '${SSO_SERVICE_ACCOUNT_NAME} ${SSO_SERVICE_ACCOUNT_CLIENT_SECRET}' < /workspace/output/cli-config/konflux/configs/pnc_cli/config.yaml > /workspace/output/config.yaml
+
+              echo -e "\nStarting the PNC build ..."
+              PNC_CLI_OUTPUT_FILE="/workspace/output/pnc-cli-run-output.json"
+              java -jar /home/jboss/bacon.jar pig run --mode=FORCE --downloadAttempts=3 /workspace/output -p /workspace/output --profile ${PNC_PROFILE} --jsonOutput > $PNC_CLI_OUTPUT_FILE
+
+              echo -e "\nFinished the PNC build!"
+              echo -e "\n=== PNC build output ==="
+              cat $PNC_CLI_OUTPUT_FILE
+              echo -e "===================="
+
+              echo -e "\nGetting the list of built files (to be used later in the container build) ..."
+              jq '[.builds[].builtArtifacts[]?.downloadUrl ]' "$PNC_CLI_OUTPUT_FILE" > "$(results.pnc-file-list.path)"
+
+              jq '{metadata: {version: "1.0"}, artifacts: [.builds[].builtArtifacts[]? | select(.downloadUrl | endswith(".zip")) | {download_url: .downloadUrl , checksums: {sha256: .sha256}}] }' $PNC_CLI_OUTPUT_FILE | yq -p json > /workspace/output/generic_lockfile.yaml
+              
+              cat  $(results.pnc-file-list.path)
+
+      workspaces:
+      - name: output
+        workspace: workspace
```

```suggestion
              jq '{metadata: {version: "1.0"}, artifacts: [.builds[].builtArtifacts[]? | select(.downloadUrl | endswith(".zip")) | {download_url: .downloadUrl , checksums: {sha256: .sha256}}] }' $PNC_CLI_OUTPUT_FILE | yq -p json > /workspace/source/generic_lockfile.yaml
              
              cat  $(results.pnc-file-list.path)

      workspaces:
      - name: output
        workspace: workspace
      - name: source
        workspace: workspace
```

This is what I meant when we talked about cachi2 looking at the source workspace